mtoken-model/src/main/java/com/wultra/security/powerauth/lib/mtoken/model/request/OperationApproveRequest.java

```diff
@@ -35,7 +39,40 @@ public class OperationApproveRequest {
     private String data;
 
     /**
-     * Optional OTP used for proximity check. User is instructed by {@link PreApprovalScreen.ScreenType#QR_SCAN}.
+     * Optional proximity check data. User is instructed by {@link PreApprovalScreen.ScreenType#QR_SCAN}.
      */
-    private String proximityCheckOtp;
+    @Schema(description = "Optional proximity check data." )
+    private ProximityCheck proximityCheck;
+
+    public Optional<ProximityCheck> getProximityCheck() {
+        return Optional.ofNullable(proximityCheck);
+    }
+
+    @Data
+    public static class ProximityCheck {
```

Comment:
Not sure if this is not long enough to justify separate class file (not a bit issue, though).

Reply:
The idea is to make a nicer API, not Java classes. When you look at the payload, it is obvious, what belongs together and what is related to the proximity check. I prefer `requestObject.proximityCheck.otp` to `requestObject.proximityCheckOtp`. 
```json
{
  "requestObject": {
    "id": "string",
    "data": "string",
    "proximityCheck": {
      "otp": "string",
      "type": "QR_CODE",
      "timestampReceived": "2023-07-18T07:56:49.516Z",
      "timestampSent": "2023-07-18T07:56:49.516Z"
    }
  }
}
```